### ISSUE TITLE
@CreateDateColumn() and @UpdateDateColumn() bug on {synchronize: true} (mssql)

### ISSUE THREAD

**Issue by periman2** [typeorm/typeorm#1991]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[x] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I found a bug when using the update and create custom columns provided by typeorm in mssql:
```
@CreateDateColumn()
    public createdAt: string;

@UpdateDateColumn()
public updatedAt: string;
```

When I add them in my entity and I enable **synchronize: true** in the db options the first time it runs okay and it created the tables. The second time and all the next one it crashes with this error:
```
.
QueryFailedError: Error: Could not create constraint or index. See previous errors.
    at new QueryFailedError (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_
modules\typeorm\error\QueryFailedError.js:27:28)
    at C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_modules\typeorm\driver
\sqlserver\SqlServerQueryRunner.js:265:61
    at _query (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_modules\mssql\
lib\base.js:1253:25)
    at Request.tds.Request.err [as userCallback] (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_modules\mssql\lib\tedious.js:620:15)
    at Request._this.callback (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\nod
e_modules\tedious\lib\request.js:60:27)
    at Connection.endOfMessageMarkerReceived (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_modules\tedious\lib\connection.js:1777:20)
    at Connection.dispatchEvent (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\n
ode_modules\tedious\lib\connection.js:863:38)
    at Parser.<anonymous> (C:\Users\FRONTEND01\Documents\Visual Studio 2017\Projects\project\project\node_mo
dules\tedious\lib\connection.js:684:18)
    at emitOne (events.js:115:13)
```
I only have two entities in my database so far and the one that I added the columns into has a OneToMany relation with the other entity. I don't know if that plays any role on resolving the issue though.

**Comment by pleerock** [comment]
Can you please specify what exact version are you using? *latest* is *0.2.0*, if you are not using this version then please upgrade to 0.2.0. If you already use 0.2.0 then provide a **minimal** reproduction git repo

**Comment by periman2** [comment]
@pleerock Sorry for the late reply. I am using the latest 0.2.0 version. I'm not sure how to create reproduction using mssql. Can I fake a db server of mssql? I'm currently using online azure database.

**Comment by et** [comment]
Hi, I'm using the latest version of 0.2.1 and this fails also on postgres:
```
'INSERT INTO "applications"("nc_uuid", "created_at", "updated_at") VALUES ($1, DEFAULT, DEFAULT) RETURNING "id", "created_at", "updated_at"',

saving entity failed { QueryFailedError: null value in column "created_at" violates not-null constraint
```

**Comment by pleerock** [comment]
simply create a minimal reproduction git repo, check it with your online azure database, make sure it does not work, then commit all your stuff without your online azure database settings, I'll use my own to check things.

Alternative it to checkout this repository, use docker to install local version of mssql, create a simple test with reproduction of this issue.

**Comment by periman2** [comment]
@pleerock Will do ASAP.

**Comment by periman2** [comment]
@pleerock I hope this is good enough: [https://github.com/periman2/typeormIssue](typeormIssue)

Please let me know if you have more questions or find out that I'm doing something wrong.

**Comment by pleerock** [comment]
Its a bug. Will be released in 0.2.3

**Comment by kgrvr** [comment]
I'm getting same error in V0.2.7:

> QueryFailedError: Error: Could not create constraint or index. See previous errors.
    at new QueryFailedError (D:\Projects\Project\Codebase\master\Project.Api\src\error\QueryFailedError.ts:7:9)
    at D:\Projects\Project\Codebase\master\Project.Api\src\driver\sqlserver\SqlServerQueryRunner.ts:209:37
    at _query (D:\Projects\Project\Codebase\master\Project.Api\node_modules\mssql\lib\base.js:1253:25)
    at Request.tds.Request.err [as userCallback] (D:\Projects\Project\Codebase\master\Project.Api\node_modules\mssql\lib\tedious.js:620:15)
    at Request._this.callback (D:\Projects\Project\Codebase\master\Project.Api\node_modules\tedious\lib\request.js:60:27)
    at Connection.endOfMessageMarkerReceived (D:\Projects\Project\Codebase\master\Project.Api\node_modules\tedious\lib\connection.js:1906:20)
    at Connection.dispatchEvent (D:\Projects\Project\Codebase\master\Project.Api\node_modules\tedious\lib\connection.js:992:38)
    at Parser.<anonymous> (D:\Projects\Project\Codebase\master\Project.Api\node_modules\tedious\lib\connection.js:813:18)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)

My problem's reproduction steps are similar to the one described above.
